### PR TITLE
Retrieve profile from request header during lep resolving

### PR DESCRIPTION
**Pull request by igordumchykov**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java [hunk 1]
```diff
@@ -17,4 +24,27 @@
...
+    protected <T> T getRequiredParam(LepMethod method, String paramName, Class<T> valueType) {
+        return ofNullable(getParamValue(method, paramName, valueType))
+            .or(() -> getHeaderValue(paramName, valueType))
+            .orElseThrow(() -> new IllegalArgumentException(
+                String.format("Neither LEP method %s nor request header contain parameter %s",
+                    getMethodDescription(method), paramName)));
+    }
+
+    private <T> Optional<T> getHeaderValue(final String headerName, final Class<T> valueType) {
+        return ofNullable((ServletRequestAttributes) RequestContextHolder.getRequestAttributes())
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java]: please consider rewriting in a functional way + avoid potential NPE in attributes.getRequest():
```
return ofNullable(RequestContextHolder.getRequestAttributes())
            .map(ServletRequestAttributes.class::cast)
            .map(ServletRequestAttributes::getRequest)
            .map(request -> request.getHeader(headerName))
            .map(valueType::cast);
```

File: src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java [hunk 2]
```diff
@@ -17,4 +24,27 @@
...
+            .map(attributes -> attributes.getRequest().getHeader(headerName))
+            .map(valueType::cast);
+    }
+
+    private String getMethodDescription(LepMethod method) {
+        MethodSignature methodSignature = method.getMethodSignature();
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java]: please consider rewriting method body in a functional way: 
```
ofNullable(method.getMethodSignature())
            .map(MethodSignature::getMethod)
            .map(Method::toString)
            .orElseGet(method::toString);
```

File: src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java (hunk 1 above)
Comment (igordumchykov) [reply src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java]: Done

File: src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java (hunk 2 above)
Comment (igordumchykov) [reply src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java]: Done

**Pull request merged**